compiler/luci/lang/include/luci/IR/CircleShapeSignature.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_IR_SHAPE_SIGNATURE_H__
+#define __LUCI_IR_SHAPE_SIGNATURE_H__
+
+#include <stdint.h>
+#include <vector>
+
+namespace luci
+{
+
+class ShapeSignature
+{
+public:
+  ShapeSignature() = default;
+
+  ShapeSignature(const std::vector<int32_t> &shape_signature)
+  {
+    _shape_signature = shape_signature;
+  }
+
+  const std::vector<int32_t> as_vector() const { return _shape_signature; }
```

Comment:
Will this return a copy of `_shape_signature` ? if so, is this your intenstion?

Reply:
This function is for following (draft)code in #4685 so it was my intension :)

https://github.com/Samsung/ONE/blob/b11505e89ac277f4177d913d9bdd4de24af3a5bf/compiler/luci/export/src/CircleTensorExporter.cpp#L256

Reply:
As it returns `const` and it's const method, caller cannot modify it. I think it would be better to return reference for performance.

Reply:
As we talked off-line, I think `std::vector<int32_t> as_vector() { ... }` can hide internal implementation and caller can R/W vector :)